dapp/src/components/buySell/SwapHomepage.js

```diff
@@ -611,9 +640,69 @@ const SwapHomepage = ({
           </a>
           <button
             //disabled={formHasErrors || buyFormHasWarnings || !totalOUSD}
+            className={`btn-blue pl-2 pr-2 buy-button mt-2 mt-md-0 w-100 ${
```

Comment:
There is quite a lot of logic in the approve button component, could we separate that into another component? The SwapHomepage is already too big -> which is something I am guilty of and should make it smaller in the future.

Reply:
Made SwapButton component.